pkg/domain/domain.go

```diff
@@ -548,6 +548,24 @@ func (do *Domain) tryLoadSchemaDiffs(m *meta.Meta, usedVersion, newVersion int64
 		}
 		diffs = append(diffs, diff)
 	}
+
+	failpoint.Inject("MockTryLoadDiffError", func(val failpoint.Value) {
+		switch val.(string) {
+		case "exchangepartition":
```

Comment:
Do we need consider other scenarios?

Reply:
Yes, but we can add more tests after we find something.
This PR is mainly for https://github.com/pingcap/tidb/issues/54796